elfpy/math/fixed_point.py

```diff
@@ -459,17 +563,17 @@ def __str__(self) -> str:
         r"""Cast to str"""
         if self.special_value is not None:
             return self.special_value
-        integer = str(self.int_value)[:-18]  # remove right-most 18 digits for whole number
+        integer = str(self.scaled_value)[:-18]  # remove right-most 18 digits for whole number
```

Comment:
I did it this way to avoid converting to float, which could potentially introduce noise. The variable `self.scaled_value` is an int type and holds the full integer + remainder precision. We know all digits left of the right-most 18 are the integer portion. Although, now that you bring this up, I see that `18` should be `self.decimal_places`.